src/client/connect.rs

```diff
@@ -42,23 +46,31 @@ where T: Service<Request=Uri, Error=io::Error> + 'static,
 
 /// A connector for the `http` scheme.
 #[derive(Clone)]
-pub struct HttpConnector {
-    dns: dns::Dns,
+pub struct HttpConnector<E = CpuPool> {
```

Comment:
I'm slightly worried about exposing the generic here. I suppose with a default set, it should still compile for everyone... The alternative is to keep a `Box<Executor>` (or perhaps `Arc<Executor>`).

Reply:
Since `HttpConnectExecutor` is wrapping `E` in an `Arc` in any case, there shouldn't be an issue with hiding the generic in there as a trait object. I'll give that a shot!

- [ ] Hide `Executor` generic